Code/Mantid/Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -644,6 +679,39 @@ MatrixWorkspace_sptr ReflectometryReductionOne::transmissonCorrection(
 }
 
 /**
+ * Perform Transmission Corrections Using Polynomials.
+ * @param IvsLam : Run workspace which is to be normalized by the results of the
+ * transmission corrections.
+ * @return Normalized run workspace by the generated polynomial.
+ */
+MatrixWorkspace_sptr
+ReflectometryReductionOne::polynomialCorrection(MatrixWorkspace_sptr IvsLam) {
+
+  const std::string corrAlgName = getProperty("CorrectionAlgorithm");
+
+  const std::string polynomial = getProperty("Polynomial");
+  const std::string c0 = getProperty("C0");
+  const std::string c1 = getProperty("C1");
```

Comment:
Is there a reason to translate the property values from their actual types to strings and then translate them back in ```corrAlg```? Are you sure no accuracy is lost?